### ISSUE TITLE
pnpm probes https urls for hosted git dependencies with retries even if response is a redirect

### ISSUE THREAD

**Issue by atsikov** [pnpm/pnpm#4074]
After adding a dependency from our bitbucket I noticed a significant increase in installation time. Turns out, before installing ssh dependency, pnpm probes https url 3 timers with increasing timeouts.

In our case, such url (something like https://bitbucket.org/company/my-package) returns a redirect to sign-in page, and fetch fails because of `redirect: 0` passed to options. Though it's not a network failure, two more fetch attempts are performed, with increasing intervals (10 seconds and 1 minute). And only after the third request fails, the dependency is fetched via correct ssh url.

### Code to reproduce the issue:
Any dependency from private git repo
```
"dependencies" : {
  "my-package": "ssh://git@bitbucket.org:company/my-package#v1.2.3"
}
```

### Expected behavior:
Dependency is installed from `ssh://` url after the first HEAD request fails

### Actual behavior:
Dependency is installed in more than a minute, only after third HEAD request fails

### Additional information:

 - `node -v` prints:v16.13.1
 - Windows, macOS, or Linux?: macOS

**Comment by zkochan** [comment]
It is maybe the same issue as this? https://github.com/pnpm/pnpm/pull/2731

**Comment by atsikov** [comment]
Yes, looks like it is. I was searching by "max-retries" error so didn't see that PR.

**Comment by zkochan** [comment]
This no longer reproduces on current pnpm, so closing.

Verified while checking 2352: pnpm hands a git specifier straight to `git ls-remote` and reports git's own error. There is no https probe with retries ahead of it.

If you still hit this on the latest pnpm, please comment with a reproduction and we will reopen.

---
Written by an agent (Claude Code, claude-opus-5).
